### PR TITLE
JOE-14: Prototype site header

### PR DESCRIPTION
**Pull request by martensc**
## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Github Issue**

- N/A

**Jira Ticket**

- [JOE-14: Prototype site header](https://palantir.atlassian.net/browse/JOE-14)

## Description

This ticket is for creating the site header component which will be used for each content type.

## To Test
1. Navigate to the [VC Header component](http://localhost:3000/?p=organisms-vc-header) and review the markup, javascript, scss, and json
    - [x] Verify the component looks correct when closed at all viewports
    - [x] Verify the "Menu" and "Search" buttons open and close when clicking on each icon/button
    - [x] Verify you can toggle between open dropdowns by clicking from the "Menu" to "Search" and vice versa
    - [x] Verify the menu and search dropdowns look good at all viewports
    - [x] Verify the "Menu" and "Search" buttons and dropdowns are navigable using your keyboard **(a11y)**
        - [x] Verify the `Escape` key closes the dropdown when open
        - [x] Verify a "Close Dropdown" button appears as you navigate through the dropdown using your keyboard
    - [x] Enable "Reduce motion" in your OS and verify all menu animations have been disabled
        - [x] On a Mac, go to "System Preferences" -> "Accessibility" -> "Display" -> Click "Reduce motion" in the listing and then refresh the component

## Visual Regressions

N/A

## Relevant Screenshots/GIFs

vc-header

## Remaining Tasks

N/A

## Additional Notes

N/A

### CODE REVIEW

**Comment by martensc** [comment]
This PR is ready for review. Don't merge until PR #144 has been merged.

**Review by jisung594 — APPROVED** [review]
Confirming that header looks and functions as instructed.

**Pull request merged**
